code/_onclick/hud/parallax.dm

```diff
@@ -0,0 +1,288 @@
+
+/obj/screen/plane_master/parallax_master
+	plane = PLANE_SPACE_PARALLAX
+	blend_mode = BLEND_MULTIPLY
+	color = list(
+	1, 0, 0, 0,
+	0, 1, 0, 0,
+	0, 0, 1, 0,
+	0, 0, 0, 0,
+	0, 0, 0, 1)
+
+/obj/screen/plane_master/parallax_spacemaster // Turns space-turfs into white. The parallax plane multiplies itself by the white, causing the parallax to only show in areas with opacity.
+	plane = PLANE_SPACE_BACKGROUND
+	color = list(
+	0, 0, 0, 0,
+	0, 0, 0, 0,
+	0, 0, 0, 0,
+	1, 1, 1, 1,
+	0, 0, 0, 0)
+
+/obj/screen/plane_master/parallax_dustmaster
+	plane = PLANE_SPACE_DUST
+
+/datum/hud/proc/update_parallax()
+	var/client/C = mymob.client
+	if(!parallax_initialized || C.updating_parallax) return
+
+	for(var/turf/space/T in range(get_turf(C.eye),C.view))
+		C.updating_parallax = 1
+		break
+
+	if(!C.updating_parallax)
+		return
+
+	//multiple sub-procs for profiling purposes
+	if(update_parallax1())
+		update_parallax2(0)
+		update_parallax3()
+		C.updating_parallax = 0
+	else
+		C.updating_parallax = 0
+
+/datum/hud/proc/update_parallax_and_dust()
+	var/client/C = mymob.client
+	if(!parallax_initialized || C.updating_parallax) return
+	C.updating_parallax = 1
+	if(update_parallax1())
+		update_parallax2(1)
+		update_parallax3()
+		C.updating_parallax = 0
+	else
+		C.updating_parallax = 0
+
+/datum/hud/proc/update_parallax1()
+	var/client/C = mymob.client
+	//DO WE UPDATE PARALLAX
+	if(C.prefs.space_parallax)//have to exclude Centcom so parallax doens't appear during hyperspace
+		parallax_on_clients |= C
+	else
+		for(var/obj/screen/parallax/bgobj in C.parallax)
+			C.screen -= bgobj
+		parallax_on_clients -= C
+		C.screen -= C.parallax_master
+		C.screen -= C.parallax_spacemaster
+		C.screen -= C.parallax_dustmaster
+		return 0
+
+	if(!C.parallax_master)
+		C.parallax_master = new /obj/screen/plane_master/parallax_master
+	if(!C.parallax_spacemaster)
+		C.parallax_spacemaster = new /obj/screen/plane_master/parallax_spacemaster
+	if(!C.parallax_dustmaster)
+		C.parallax_dustmaster = new /obj/screen/plane_master/parallax_dustmaster
+	return 1
+
+/datum/hud/proc/update_parallax2(forcerecalibrate = 0)
+	var/client/C = mymob.client
+	//DO WE HAVE TO REPLACE ALL THE LAYERS
+
+	if(!C.parallax.len)
+		for(var/obj/screen/parallax/bgobj in parallax_icon)
+			var/obj/screen/parallax/parallax_layer = new /obj/screen/parallax
+			parallax_layer.appearance = bgobj.appearance
+			parallax_layer.base_offset_x = bgobj.base_offset_x
+			parallax_layer.base_offset_y = bgobj.base_offset_y
+			parallax_layer.parallax_speed = bgobj.parallax_speed
+			C.parallax += parallax_layer
+
+	var/parallax_loaded = 0
+	for(var/obj/screen/parallax/S in C.screen)
+		parallax_loaded = 1
+		break
+
+	if(forcerecalibrate || !parallax_loaded)
+		for(var/obj/screen/parallax/bgobj in C.parallax)
+			C.screen |= bgobj
+
+		C.screen |= C.parallax_master
+		C.screen |= C.parallax_spacemaster
+		C.screen |= C.parallax_dustmaster
+		C.parallax_dustmaster.color = list(0,0,0,0)
```

Comment:
A current bug with this is that the dustmaster shouldn't be taken out of the screen so that this has an effect